File: sort/h-index.py
def solution(citations):
    answer = 0
    n = len(citations)
    arr = [0] * (n+1)
    for c in range(n):
        now = citations[c]
        if now > n:
            now = n
        while now > 0:
            arr[now] += 1
            now -= 1
    for i in range(1, n+1):
        if arr[i] >= i:
            answer = max(answer, i)
    return answer

File: sort/h-index.js
function solution(citations) {
  let answer = 0;
  const length = citations.length;
  let max_h = 1;
  while (1) {
    let total = 0;
    for (let i = 0; i < length; i++) {
      if (citations[i] >= max_h) {
        total += 1;
      }
    }
    if (total < max_h) {
      break;
    } else {
      answer = max_h;
      max_h += 1;
    }
  }
  return answer;
}
